mesa/experimental/mesa_signals/mesa_signal.py

```diff
@@ -472,6 +472,189 @@ def _mesa_notify(self, signal: AttributeDict):
+
+        # Calculate new value based on time
+        current_time = self._get_time(instance)
+        elapsed = current_time - state.last_update
+
+        if elapsed > 0:
+            old_value = state.value
+            new_value = state.calculate(elapsed, instance)
+
+            # Check thresholds
+            crossed = state.check_thresholds(old_value, new_value)
+
+            # Update stored state
+            state.value = new_value
+            state.last_update = current_time
+
+            # Emit signals
+            if new_value != old_value:
+                instance.notify(self.public_name, old_value, new_value, "change")
+
+            for threshold, direction in crossed:
+                instance.notify(
+                    self.public_name,
+                    old_value,
+                    new_value,
+                    "threshold_crossed",
+                    threshold=threshold,
+                    direction=direction,
+                )
+
+        # Register dependency if inside a Computed
+        if CURRENT_COMPUTED is not None:
+            CURRENT_COMPUTED._add_parent(instance, self.public_name, state.value)
+            PROCESSING_SIGNALS.add(_hashable_signal(instance, self.public_name))
+
+        return state.value
+
+    # TODO: A universal truth for time should be implemented structurally in Mesa. See https://github.com/projectmesa/mesa/discussions/2228
+    def _get_time(self, instance):
+        """Get current time from model, trying multiple sources."""
+        model = instance.model
+
+        # Try simulator time first (for DEVS models)
+        if hasattr(model, "simulator") and hasattr(model.simulator, "time"):
+            return model.simulator.time
+
+        # Fall back to model.time if it exists
+        if hasattr(model, "time"):
+            return model.time
+
+        # Last resort: use steps as a proxy for time
+        return float(model.steps)
+
+
+class ContinuousState:
+    """Internal state tracker for continuous observables."""
+
+    __slots__ = ["last_update", "rate_func", "thresholds", "value"]
+
+    def __init__(self, value: float, last_update: float, rate_func: Callable):
+        self.value = value
+        self.last_update = last_update
+        self.rate_func = rate_func
+        self.thresholds = {}  # {threshold_value: set(callbacks)}
+
+    def calculate(self, elapsed: float, instance: Any) -> float:
+        """Calculate new value based on elapsed time.
+
+        Uses simple linear integration for now. Could be extended
+        to support more complex integration methods.
+        """
+        rate = self.rate_func(self.value, elapsed, instance)
+        return self.value + (rate * elapsed)
+
+    def check_thresholds(self, old_value: float, new_value: float) -> list:
+        """Check if any thresholds were crossed.
+
+        Returns:
+            List of (threshold_value, direction) tuples for crossed thresholds
+        """
+        crossed = []
```

Comment:
can you indicate whether you are only interested in moving above or moving below, or do you get any crossing of the threshold?